tests/testthat/test-object_name_linter.R

```diff
@@ -203,14 +210,28 @@ test_that("object_name_linter supports custom regexes", {
 
       myBadName <- 20L
     "),
-    list(
-      list(line_number = 1L, message = msg2),
-      list(line_number = 2L, message = msg2),
-      # argument "id" is linted if we only allow shinyModule names
-      list(line_number = 4L, column_number = 24L, message = msg2),
-      list(line_number = 8L, column_number = 28L, message = msg2),
-      list(line_number = 12L, message = msg2)
-    ),
+    list(line_number = 12L, message = msg2),
     linter2
   )
+
+  # Default regex naming works
+  expect_lint(
+    trim_some("
+      a <- 42L
+      b <- 1L
+      c <- 2L
+    "),
+    list(line_number = 3L, message = rex::rex("Variable and function name style should be custom.")),
```

Comment:
WDYT about something like:

> Variable and function name style should match custom pattern(s) %s

I think it's likely many people who read the lint message aren't authors -- the current message will lead to a confusing experience.

e.g. contributors to a package on GH shouldn't need to know any details about lintr in order to get actionable feedback if their PR is blocked by lintr.

in our case, linting is happening in an even more decentralized fashion -- lints show up in the editor and you would have to do some digging to find out what 'custom' means.

Reply:
Excellent idea. We can auto-name the regexes by themselves to achieve a more actionable message. "... should match snake_case or ^bla$" also works for me in the context of predefined styles.